### ISSUE TITLE
sample.lua fails if vocabulary has no spaces

### ISSUE THREAD

**Issue by raidancampbell** [karpathy/char-rnn#44]
After attempting to train on a [list of names from the US census](http://stackoverflow.com/questions/1803628/raw-list-of-person-names), I couldn't sample any input from the network.  It turns out that the default primetext (a space character) didn't exist in the dataset.

    /usr/local/bin/luajit: ./util/OneHot.lua:14: bad argument #1 to 'size' (dimension 1 out of range of 0D tensor at /tmp/luarocks_torch-scm-1-9151/torch7/generic/Tensor.c:17)
    stack traceback:
        [C]: in function 'size'
        ./util/OneHot.lua:14: in function 'func'
        /usr/local/share/lua/5.1/nngraph/gmodule.lua:214: in function 'neteval'
        /usr/local/share/lua/5.1/nngraph/gmodule.lua:244: in function 'forward'
        sample.lua:82: in main chunk
        [C]: in function 'dofile'
        /usr/local/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
        [C]: at 0x00405900

I'll readily admit I don't understand much of the codebase, such as why the random seed defaults to a constant value.  However, using the defaulted constant random seed should still allow a deterministic primetext character to be chosen from the vocabulary.  This should retain the deterministic output of the RNN for repeated queries.

tl;dr randomize the primetext in sample.lua using the 'seed' field to seed the randomizer to choose the primetext character.

**Comment by karpathy** [comment]
I'm confused. Are you using the latest code? I'm pretty sure this should work fine

**Comment by raidancampbell** [comment]
you're completely correct, my apologies

**Comment by karpathy** [comment]
no problem :) thank you for bringing up what you thought was an issue :) cheers!
